fe/fe-core/src/main/java/com/starrocks/staros/BDBJEJournalSystem.java

```diff
@@ -0,0 +1,195 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+package com.starrocks.staros;
+
+import com.staros.exception.ExceptionCode;
+import com.staros.exception.StarException;
+import com.staros.journal.Journal;
+import com.staros.journal.JournalSystem;
+import com.starrocks.common.Config;
+import com.starrocks.common.util.Daemon;
+import com.starrocks.common.util.Util;
+import com.starrocks.journal.JournalCursor;
+import com.starrocks.journal.JournalEntity;
+import com.starrocks.journal.JournalException;
+import com.starrocks.journal.JournalInconsistentException;
+import com.starrocks.journal.JournalTask;
+import com.starrocks.journal.JournalWriter;
+import com.starrocks.journal.bdbje.BDBEnvironment;
+import com.starrocks.journal.bdbje.BDBJEJournal;
+import com.starrocks.persist.EditLog;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.atomic.AtomicLong;
+
+// wrapper for star manager to use bdbje
+public class BDBJEJournalSystem implements JournalSystem {
+    private static final String JOURNAL_PREFIX = "starmgr_"; // do not change this string!
+    private static final int REPLAY_INTERVAL_MS = 1;
+    private static final Logger LOG = LogManager.getLogger(BDBJEJournalSystem.class);
+
+    private BDBJEJournal bdbjeJournal;
+    private JournalWriter journalWriter;
+    private EditLog editLog;
+    private AtomicLong replayedJournalId;
+    private Daemon replayer; // TODO: maybe it's better to move this to StarMgr
+
+    public BDBJEJournalSystem(BDBEnvironment environment) {
+        BlockingQueue<JournalTask> journalQueue = new ArrayBlockingQueue<JournalTask>(Config.metadata_journal_queue_size);
+        bdbjeJournal = new BDBJEJournal(environment, JOURNAL_PREFIX);
+
+        journalWriter = new JournalWriter(bdbjeJournal, journalQueue);
+
+        editLog = new EditLog(journalQueue);
+
+        replayedJournalId = new AtomicLong(0L);
+
+        replayer = null;
+    }
+
+    // for checkpoint thread only
+    public BDBJEJournalSystem(BDBJEJournal journal) {
+        bdbjeJournal = journal;
+        replayedJournalId = new AtomicLong(0L);
+    }
+
+    public long getReplayId() {
```

Comment:
getReplayedId or getReplayedJournalId are better.

Reply:
that needs to release a new star mgr version, maybe later